src/PdoConnection.php

```diff
@@ -248,7 +268,17 @@ public function rollBack()
      */
     public function commit()
     {
-        return $this->connect()->commit();
+        if (--$this->transactionCount === 0) {
```

Comment:
I could try to make it more robust for handling cases like these, but that seems like a pretty clear dev error.